src/dev/platform/f4xx/Timerf4xx.cpp

```diff
@@ -0,0 +1,252 @@
+ * @return an uint8_t between 0-9 corresponding to an element of timerInterruptHandlers
+ */
+uint8_t getTimerInterruptIndex(TIM_TypeDef* peripheral) {
+    uint8_t interruptIdx;
+
+    if (peripheral == TIM2) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM2_IDX);
+    } else if (peripheral == TIM3) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM3_IDX);
+    } else if (peripheral == TIM4) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM4_IDX);
+    } else if (peripheral == TIM5) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM5_IDX);
+    } else if (peripheral == TIM9) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM9_IDX);
+    } else if (peripheral == TIM10) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM10_IDX);
+    } else if (peripheral == TIM11) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM11_IDX);
+    } else if (peripheral == TIM12) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM12_IDX);
+    } else if (peripheral == TIM13) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM13_IDX);
+    } else if (peripheral == TIM14) {
+        interruptIdx = static_cast<uint8_t>(timerInterruptIndex::TIM14_IDX);
+    } else {
+        interruptIdx = -1;
+    }
+
+    return interruptIdx;
+}
+
+namespace EVT::core::DEV {
+
+Timerf4xx::Timerf4xx(TIM_TypeDef* timerPeripheral, uint32_t clockPeriod) {
+    initTimer(timerPeripheral, clockPeriod);
+    this->halTimer = &halTimers[getTimerInterruptIndex(timerPeripheral)];
+}
+
+void Timerf4xx::initTimer(TIM_TypeDef* timerPeripheral, uint32_t clockPeriod) {
+    this->clockPeriod = clockPeriod;
+    auto& htim = halTimers[getTimerInterruptIndex(timerPeripheral)];
+
+    htim.Instance = timerPeripheral;
+    uint32_t prescaler = SystemCoreClock / 1000;
+    htim.Init.Prescaler = prescaler;// Sets f_CK_PSC to 1000 Hz
+    // Allows period increments of 1 ms with max of 2^(32) ms.
+    htim.Init.CounterMode = TIM_COUNTERMODE_UP;
+    htim.Init.Period = clockPeriod;
+    htim.Init.ClockDivision = TIM_CLOCKDIVISION_DIV1;
+    htim.Init.AutoReloadPreload = TIM_AUTORELOAD_PRELOAD_ENABLE;
+    HAL_TIM_Base_Init(&htim);
+
+    TIM_ClockConfigTypeDef clockConfig = {0};
+    TIM_MasterConfigTypeDef masterConfig = {0};
+
+    clockConfig.ClockSource = TIM_CLOCKSOURCE_INTERNAL;
+    HAL_TIM_ConfigClockSource(&htim, &clockConfig);
+
+    // Timers 9-14 are NOT master mode compatible, so waste of time to go through config
+    if (getTimerInterruptIndex(timerPeripheral) < static_cast<uint8_t>(timerInterruptIndex::TIM9_IDX)) {
+        masterConfig.MasterOutputTrigger = TIM_TRGO_RESET;
+        masterConfig.MasterSlaveMode = TIM_MASTERSLAVEMODE_DISABLE;
+        HAL_TIMEx_MasterConfigSynchronization(&htim, &masterConfig);
+    }
+}
+
+void Timerf4xx::startTimer(void (*irqHandler)(void* htim)) {
```

Comment:
In start time you run stop timer. Lowkey, it might be more efficient to first check if the timer is actually running before stopping it to avoid redundant operations.